packages/js/data/src/products/resolvers.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * External dependencies
+ */
+import { addQueryArgs } from '@wordpress/url';
+import { apiFetch } from '@wordpress/data-controls';
+
+/**
+ * Internal dependencies
+ */
+import { WC_PRODUCT_NAMESPACE } from './constants';
+import { setError, setProducts, setProductsTotalCount } from './actions';
+import { Product, ProductQuery } from './types';
+import { fetchWithHeaders } from '../controls';
+
+type ProductGetResponse = Product[] | ( { data: Product[] } & Response );
+
+function* request( query: Partial< ProductQuery > ) {
```

Comment:
This looks very similar to the `items` data store request method.  Any opportunities to dry up some of this logic or are they too unique?

Reply:
They should be near identical aside from the types, but I can make use of a generic for that.